code/game/machinery/recharger.dm

```diff
@@ -146,6 +149,11 @@
 				using_power = TRUE
 			update_appearance()
 			return
+		if(!using_power && !finished_recharging) //Inserted thing is at max charge/ammo, notify those around us
+			finished_recharging = TRUE
+			playsound(src, 'sound/machines/ping.ogg', 30, TRUE)
+			say("[charging] has finished recharging!")
```

Comment:
Should we use bubble messages instead of say and the chat window?

Reply:
Toss-up. Archival isn't important, but this is something that might exist as a real message in-world.

I think in my mind, the ideal version of this is a say that doesn't write in chat, but we don't have that capability yet.

Reply:
What is the difference between a bubble message, and a say that doesn't write in chat? Deafness checks?

Reply:
Balloon alerts fade much faster, and say messages imply that it is being said in-world.

For example--a balloon alert for "gears set to clamp" makes sense, but a say message for it would not.

Reply:
Maintainer Tralezab here, I think a balloon message would be best. Machines talking is always cool, but this machine is a glorified battery charger. Of course it really doesn't matter in the end and I think both are fine